src/KnowledgeQuiz.tsx

```diff
@@ -42,35 +53,104 @@ const fieldsetStyle = css`
 export const KnowledgeQuizAtom = ({
     id,
     questions,
-}: QuizAtomType): JSX.Element => (
-    <form data-atom-id={id}>
-        {questions.map((question, idx) => (
-            <Question
-                key={question.id}
-                id={question.id}
-                number={idx + 1}
-                text={question.text}
-                imageUrl={question.imageUrl}
-                answers={question.answers}
-            />
-        ))}
-    </form>
-);
+    resultGroups,
+    sharingIcons,
+}: QuizAtomType): JSX.Element => {
+    const [selectedGlobalAnswers, setSelectedGlobalAnswers] = useState<{
+        [questionId: string]: AnswerType;
+    }>({});
+
+    const [
+        haveAllQuestionsBeenAnswered,
+        setHaveAllQuestionsBeenAnswered,
+    ] = useState<boolean>(false);
+
+    useEffect(() => {
+        if (Object.keys(selectedGlobalAnswers).length === questions.length) {
+            setHaveAllQuestionsBeenAnswered(true);
+        }
+    }, [selectedGlobalAnswers, setHaveAllQuestionsBeenAnswered, questions]);
+
+    return (
+        <form data-atom-id={id}>
+            {haveAllQuestionsBeenAnswered && (
+                <div data-testid="quiz-results-block-top">
+                    <Result
+                        selectedGlobalAnswers={selectedGlobalAnswers}
+                        resultGroups={resultGroups}
+                        sharingIcons={sharingIcons}
+                    />
+                </div>
+            )}
+            {questions.map((question, idx) => (
+                <Question
+                    key={question.id}
+                    id={question.id}
+                    number={idx + 1}
+                    text={question.text}
+                    imageUrl={question.imageUrl}
+                    answers={question.answers}
```

Comment:
yes, thought with the renaming I hope it will be a little clearer, it essentually list the options each question can have. We would map each question selection to quizSelection (what I previously referred to as globalAnswers).